firebase_admin/tenant_mgt.py

```diff
@@ -0,0 +1,445 @@
+TenantNotFoundError = _auth_utils.TenantNotFoundError
+
+
+def auth_for_tenant(tenant_id, app=None):
+    """Gets an Auth Client instance scoped to the given tenant ID.
+
+    Args:
+        tenant_id: A tenant ID string.
+        app: An App instance (optional).
+
+    Returns:
+        auth.Client: An ``auth.Client`` object.
+
+    Raises:
+        ValueError: If the tenant ID is None, empty or not a string.
+    """
+    tenant_mgt_service = _get_tenant_mgt_service(app)
+    return tenant_mgt_service.auth_for_tenant(tenant_id)
+
+
+def get_tenant(tenant_id, app=None):
+    """Gets the tenant corresponding to the given ``tenant_id``.
+
+    Args:
+        tenant_id: A tenant ID string.
+        app: An App instance (optional).
+
+    Returns:
+        Tenant: A Tenant object.
+
+    Raises:
+        ValueError: If the tenant ID is None, empty or not a string.
+        TenantNotFoundError: If no tenant exists by the given ID.
+        FirebaseError: If an error occurs while retrieving the tenant.
+    """
+    tenant_mgt_service = _get_tenant_mgt_service(app)
+    return tenant_mgt_service.get_tenant(tenant_id)
+
+
+def create_tenant(
+        display_name, allow_password_sign_up=None, enable_email_link_sign_in=None, app=None):
+    """Creates a new tenant from the given options.
+
+    Args:
+        display_name: Display name string for the new tenant. Must begin with a letter and contain
+            only letters, digits and hyphens. Length must be between 4 and 20.
+        allow_password_sign_up: A boolean indicating whether to enable or disable the email sign-in
+            provider (optional).
+        enable_email_link_sign_in: A boolean indicating whether to enable or disable email link
+            sign-in (optional). Disabling this makes the password required for email sign-in.
+        app: An App instance (optional).
+
+    Returns:
+        Tenant: A Tenant object.
+
+    Raises:
+        ValueError: If any of the given arguments are invalid.
+        FirebaseError: If an error occurs while creating the tenant.
+    """
+    tenant_mgt_service = _get_tenant_mgt_service(app)
+    return tenant_mgt_service.create_tenant(
+        display_name=display_name, allow_password_sign_up=allow_password_sign_up,
+        enable_email_link_sign_in=enable_email_link_sign_in)
+
+
+def update_tenant(
+        tenant_id, display_name=None, allow_password_sign_up=None, enable_email_link_sign_in=None,
+        app=None):
+    """Updates an existing tenant with the given options.
+
+    Args:
+        tenant_id: ID of the tenant to update.
+        display_name: Updated display name string for the tenant (optional).
+        allow_password_sign_up: A boolean indicating whether to enable or disable the email sign-in
+            provider.
+        enable_email_link_sign_in: A boolean indicating whether to enable or disable email link
+            sign-in. Disabling this makes the password required for email sign-in.
```

Comment:
Is it really disabling it?  That seems mildly counterintuitive.

Reply:
Setting it to False disables the feature